VideoWeb/VideoWeb/ClientApp/src/app/shared/error/error.component.ts

```diff
@@ -23,43 +25,77 @@ export class ErrorComponent implements OnInit, OnDestroy {
     errorMessageBody: string;
     connectionError: boolean;
     showReconnect: boolean;
+    attemptingReconnect: boolean;
 
-    constructor(private router: Router, private pageTracker: PageTrackerService, private eventsService: EventsService) {
+    constructor(
+        private router: Router,
+        private pageTracker: PageTrackerService,
+        private eventsService: EventsService,
+        private logger: Logger
+    ) {
         this.browserRefresh = false;
-        this.subscription = this.router.events.subscribe(event => {
-            if (event instanceof NavigationEnd) {
-                this.browserRefresh = event.id === 1 && event.url === event.urlAfterRedirects;
-            }
+        this.checkForRefresh();
+    }
 
-            if (this.browserRefresh) {
-                this.goBack();
-            } else {
-                this.startGoBackTimer();
-            }
-        });
+    get hasInternetConnection(): boolean {
+        return window.navigator.onLine;
     }
 
     ngOnInit(): void {
-        this.eventsService.stop();
+        this.attemptingReconnect = false;
+        if (this.eventsService.isConnectedToHub) {
+            this.eventsService.stop();
+        }
```

Comment:
should this be here? the event service should be responsible for handling the connection state?

Reply:
moved to events service